File: ARCHISOFT_topoalign/DialogUserConfirmation.vb
Imports System.Windows.Forms
Imports Microsoft.AppCenter.Crashes

Public Class DialogUserConfirmation

    Private _clickResult As UserConfirmation
    Public Property ClickResult() As UserConfirmation
        Get
            Return _clickResult
        End Get
        Set(ByVal value As UserConfirmation)
            _clickResult = value
        End Set
    End Property

    Private Sub OK_Button_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles OK_Button.Click
        ClickResult = UserConfirmation.Send
        Me.DialogResult = DialogResult.None
        Me.Close()
    End Sub

    Private Sub Cancel_Button_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Cancel_Button.Click
        ClickResult = UserConfirmation.DontSend
        Me.DialogResult = DialogResult.None
        Me.Close()
    End Sub

End Class

File: TopoAlign/DialogUserConfirmation.cs
using Microsoft.AppCenter.Crashes;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace TopoAlign
{
    public partial class DialogUserConfirmation : Form
    {
        public UserConfirmation ClickResult { get; set; }

        public DialogUserConfirmation()
        {
            InitializeComponent();
        }

        private void OK_Button_Click(object sender, EventArgs e)
        {
            ClickResult = UserConfirmation.Send;
            this.DialogResult = DialogResult.None;
            this.Close();
        }

        private void Cancel_Button_Click(object sender, EventArgs e)
        {
            ClickResult = UserConfirmation.DontSend;
            this.DialogResult = DialogResult.None;
            this.Close();
        }
    }
}
